src/lib/DiscordBot/DiscordBot.ts

```diff
@@ -136,11 +145,9 @@ export class DiscordBot {
     const payload: CreateChatCompletionPayloadMessage[] = [];
     payload.push(await this._constructSystemPrompt(systemPromptOverride));
 
-    this._messageHistory.forEach(async message => {
-      if (message.convoKey === convoKey) {
-        payload.push(message.payload);
-      }
-    });
+    this._historyMessageBucket.objects
```

Comment:
Nothing is done with the result of this. Needs to be wrapped with `payload.concat()`

Reply:
Committed fix